paraatm/simulation_method/vcas.py

```diff
@@ -0,0 +1,380 @@
+"""
+NASA NextGen NAS ULI Information Fusion
+
+@organization: Arizona State University
+@author: Yuhao Wang
+@date: 2020-04-19
+
+This Python script is used for VCAS (Voice Communication-Assisted Simulation)
+Currently monitors altitude related commands during descend only
+The script outputs aircraft trajectory based on input flight plan and text command translated from audio
+Simulation based on NATS beta1.7 standalone version
+The script can also check compliance for each command
+"""
+
+import time
+import os
+import io
+from shutil import copyfile
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import scipy.stats as sts
+
+from paraatm.io.nats import NatsSimulationWrapper, NatsEnvironment, read_nats_output_file
+
+
+class VCAS(NatsSimulationWrapper, object):
+    """Class method for VCAS simulation and calculation based on input
+    """
+    def __init__(self, cfg):
+        """
+        extract input as self variable
+        initialize NATS standalone server, retrieve NATS interfaces as self variable
+        get simulation constant
+
+        Parameters
+        ----------
+        cfg : dictionary
+            inputs to VCAS simulation
+            'fp_file' :  str
+                directory tp flight plan file for NATS simulation
+            'mfl_file' : str
+                directory to mfl file for NATS simulation
+            'cmd_file' : str
+                directory to text command as .csv file
+            'data_file' : srt
+                directory to actual trajectory data as .csv file
+            'sim_time' : int/float
+                total simulation time
```

Comment:
The indentation is wrong for parameters 2-6.

Reply:
Listed from 2 to 6 is the inputs that should be included in the dictionary cfg. I am adding the indent to indicate that this items should be inside the input dictionary.